### PR TITLE
ADAPT-1178 ADAPT-1785 ADAPT-1789: Alert Component and Misc Storybook Config

### PR DESCRIPTION
**Pull request by sherakama**
# READY FOR REVIEW

# Summary
- Alert & Button Component
- Storybook configuration and additional add ons.
- Some default HTML elements added as stories.
- Beginning of jest tests

# Needed By (Date)
- End of the sprint

# Urgency
- Normal
 
# Steps to Test

1. Check out the preview build and try out all of the different control options on the alert component
2. Compare the layouts in the Figma mock to the options provided and record any gaps. 
3. Review styles and responsive behavior for a couple of configurations of levers
4. Review code 
5. Review accessibility / Browser Compatibility
6. Review the scripts section in package.json and run each one to get familiar. 

# Affected Projects or Products
- Decanter, ADAPT

# Associated Issues and/or People
- ADAPT-1178, ADAPT-1785, ADAPT-1789
- @katrialesser 

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: .storybook/main.js
Comment (sherakama) [inline .storybook/main.js:5]: Added this as a place we can put stories we don't want people to use directly but would like to have in the storybook build.

File: .storybook/stories/Blockquote.stories.js
Comment (sherakama) [inline .storybook/stories/Blockquote.stories.js:1]: This story is here to show the default HTML element of Decanter.

File: .storybook/stories/Heading.stories.js
Comment (sherakama) [inline .storybook/stories/Heading.stories.js:1]: This story is here to show the default HTML element styles of headings in Decanter.

File: .gitignore
Comment (sherakama) [inline .gitignore:45]: This stores the results of the jest tests and is used to display them in the `tests` tab in the storybook ui.

File: .storybook/main.js
Comment (sherakama) [inline .storybook/main.js:11]: For running/supporting tests.

File: .storybook/main.js
Comment (sherakama) [inline .storybook/main.js:23]: Handy plugin to help with states like hover/focus.

File: .storybook/preview.js
Comment (sherakama) [inline .storybook/preview.js:10]: Added some of our breakpoints and a misc height for testing.

File: .storybook/preview.js
Comment (sherakama) [inline .storybook/preview.js:64]: Added our default color palette for background options to use in the storybook ui.

File: .storybook/preview.js
Comment (sherakama) [inline .storybook/preview.js:160]: This adds the `tests` tab using the data from the json file.

File: .storybook/stories/List.stories.js
Comment (sherakama) [inline .storybook/stories/List.stories.js:1]: A story to showcase default list styles.

File: .storybook/stories/Paragraph.stories.js
Comment (sherakama) [inline .storybook/stories/Paragraph.stories.js:1]: A story to showcase default paragraph styles and a few levers.

File: babel.config.js
Comment (sherakama) [inline babel.config.js:6]: This is needed to get jest to run.

File: dist/index.css
Comment (sherakama) [inline dist/index.css:1]: Removed all of the create-react-app generations in favor of storybook.

File: package.json
Comment (sherakama) [inline package.json:8]: No more bundler output. Just use our source.

File: package.json
Comment (sherakama) [inline package.json:20]: Got rid of the bundler and github pages stuff that we were not using anhow.

File: package.json
Comment (sherakama) [inline package.json:49]: Bye bye.

File: package.json
Comment (sherakama) [inline package.json:40]: This is a dependency as there is an import in a component.

File: package.json
Comment (sherakama) [inline package.json:70]: Upgraded to react 17!

**Comment by yvonnetangsu** [comment]
Whoa someone's been busy 😄

File: src/Alert/Alert.js
Comment (sherakama) [inline src/Alert/Alert.js:15]: moved `ref` to the props as forwardRef makes it required and I don't want that here.

Comment (sherakama) [reply src/Alert/Alert.js:15]: This component got complicated again and has a fair bit of logic in it to support all the levers and a couple of combinations. We're going to have to be mindful of this as we go and to break down the components into separate variant files if the logic gets too difficult or complicated.

File: src/Alert/Alert.stories.js
Comment (sherakama) [inline src/Alert/Alert.stories.js:32]: If you put these options on the default export they are considered 'global' for the component.

File: src/Alert/Alert.stories.js
Comment (sherakama) [inline src/Alert/Alert.stories.js:53]: We will have to talk about our test coverage and if we want to have one global test file or to break them down into per-story test files.

File: src/Alert/Alert.stories.js
Comment (sherakama) [inline src/Alert/Alert.stories.js:69]: This is how you get text above the example/documentation on the `docs` pages per story.

File: src/Alert/Alert.test.js
Comment (sherakama) [inline src/Alert/Alert.test.js:1]: I am just getting started with jest and am using this to test that it can be run and see how it integrates with storybook.

File: src/Alert/Alert.js
Comment (sherakama) [inline src/Alert/Alert.js:4]: The react-aria library button was a hidden button for screen readers only so I did not use that one. We needed a button component so I started one.

File: src/Button/Button.js
Comment (sherakama) [inline src/Button/Button.js:7]: Start of a button component. I suspect we will have a number of levers (big, etc) later.

File: src/StyledLink/index.js
Comment (sherakama) [inline src/StyledLink/index.js:12]: Removing all of this and will recreate it with hero icons later.
Also, thing to note, jest doesn't like css modules and we'd have to add extra support for that.

**Comment by sherakama** [comment]
Ready for review.

**Comment by sherakama** [comment]
When you test locally you will likely want to delete your `package-lock.json` and `node_modules` directory and install as the packages have shifted a fair bit.

**Comment by sherakama** [comment]
> Whoa someone's been busy 😄

It is not often I get a whole heads-down day 👍

**Comment by yvonnetangsu** [comment]
@sherakama I deleted my `node_modules` folder (and did a `npm cache clean --force` just in case) and deleted my `package-lock.json`. When I do `npm install`, it gave me a bunch of errors. However, I changed the react, react-dom and react-test-renderer version to `^16.14.0` and it installs without error (and also a lot less warning) 🤔

**Comment by yvonnetangsu** [comment]
When using react/react-dom/react-test-renderer 17.0.1, I got these errors for doing `npm install`. But they go away if I switch back to version 16.14.0
Screen Shot 2021-02-16 at 9 45 48 AM

**Comment by sherakama** [comment]
> When using react/react-dom/react-test-renderer 17.0.1, I got these errors for doing `npm install`. But they go away if I switch back to version 16.14.0
> Screen Shot 2021-02-16 at 9 45 48 AM

I think you have some legacy information in there somewhere. The preview site builds without issue so I think you need a clean build.

Try these steps in order.
1. delete node_modules
2. delete package-lock.json
3. npm install
4. npm run build

**Comment by sherakama** [comment]
All in all, we could probably go back to version 16 as that might help with overall compatibility.

**Comment by yvonnetangsu** [comment]
> All in all, we could probably go back to version 16 as that might help with overall compatibility.

Yeah I did all those steps and even did a `npm cache clean --force`. Let me try one more time.

**Comment by yvonnetangsu** [comment]
@sherakama Yeah, still got the same error. I even tried updating my `npm`. Hmm...could we go back to 16 for now?

**Comment by sherakama** [comment]
> @sherakama Yeah, still got the same error. I even tried updating my `npm`. Hmm...could we go back to 16 for now?

Yeah, we could probably go with something like `>=16.14.0`

File: package.json
Comment (yvonnetangsu) [inline package.json:73]:
```suggestion
    "react-test-renderer": "^16.14.0",
```

File: package.json
Comment (yvonnetangsu) [inline package.json:72]:
```suggestion
    "react": "^16.14.0",
    "react-dom": "^16.14.0",
```

**Comment by sherakama** [comment]
Try that. We can support version 16 and 17 in this package.

**Comment by yvonnetangsu** [comment]
> Try that. We can support version 16 and 17 in this package.

Yikes, that surprisingly didn't work for me either (I did the whole deleting everything step too). Seems like the only thing that would work for me is ^16.14.0 🤷🏼

**Comment by yvonnetangsu** [comment]
@sherakama I pushed up the version changes to ^16.14.0 and some minor style tweaks.

File: .storybook/preview.js
Comment (yvonnetangsu) [reply .storybook/preview.js:10]: This is nice

File: src/StyledLink/index.js
Comment (yvonnetangsu) [inline src/StyledLink/index.js:45]: Nice - thanks for starting work on these also

File: src/Alert/Alert.js
Comment (yvonnetangsu) [inline src/Alert/Alert.js:209]: It's interesting you did it this way - nice that it allows for some flexibility here.

File: src/Alert/Alert.levers.js
Comment (yvonnetangsu) [inline src/Alert/Alert.levers.js:11]: These probably need focus: classes here but I see that you're adding some stuff to decanter-7 so can wait till those are merged into v7 branch.

File: src/Alert/Alert.stories.js
Comment (yvonnetangsu) [reply src/Alert/Alert.stories.js:53]: Let's talk more in our adapt dev chat. Per story seems like a nice idea so everything is modular, but if only a global one is less work I'm fine with that too.

File: src/Button/Button.js
Comment (yvonnetangsu) [inline src/Button/Button.js:34]: Looks like we can attach an action to this:
https://storybook.js.org/docs/react/essentials/actions

**Comment by sherakama** [comment]
If you think it is GTG then this is GTG. Please approve and merge when you are ready.

**Review by yvonnetangsu — APPROVED** [review]
Thank you for the huge amount of work you did setting this up. I like how things are structured. The booleans for markup changes are easy enough to understand and the logic is not nearly as convoluted as our old Twig templates. It seems like this would work well with passing things in through CMS - looks like you provide a lot of flexibility there.

I think this is ready to be merged in and we can continue to build off of this and make small changes if needed. We can use this to start testing with Storyblok too.

**Comment by yvonnetangsu** [comment]
🚀 🎸

**Pull request merged**
